### PR TITLE
Add DotNetCliTool package verification

### PR DESCRIPTION
**Pull request by natemcmaster**
Opt-out of xml rule for DotNetCliTool packages.

We had some issues with our CLI packages in recent builds due to bugs in NuGet pack. This should help ensure layout

cc @ajaybhargavb - fixed some NREs and added '--help'
cc @pranavkm @bricelam - we can simplify our NugetPackageVerifier.json files so we dont' have to exempt our packages from the MISSING_DOCs rule

### CODE REVIEW

File: src/NuGetPackageVerifier/Program.cs [hunk 1]
```diff
@@ -75,6 +74,10 @@ public static int Main(string[] args)
 
                 logger.LogNormal("Read {0} package set(s) from {1}", packageSets.Count, packagesToScanJsonFilePath);
             }
+            else
+            {
+                packageSets = new Dictionary<string, PackageSet>();
```
Comment (ajaybhargavb) [inline src/NuGetPackageVerifier/Program.cs]: Why is this needed?

File: src/NuGetPackageVerifier/CompositeRules/DefaultCompositeRule.cs [hunk 2]
```diff
@@ -23,6 +23,7 @@ public class DefaultCompositeRule : IPackageVerifierRule
             new SatellitePackageRule(),
             new StrictSemanticVersionValidationRule(),
             new DependenciesVersionRangeBoundsRule(),
+            new DotNetCliToolPackageRule(),
```
Comment (ajaybhargavb) [inline src/NuGetPackageVerifier/CompositeRules/DefaultCompositeRule.cs]: I know we don't use any other rule list, but it would be nice to add new rules to other lists like `allRules`. It helps with debugging.

File: src/NuGetPackageVerifier/Program.cs (hunk 1 above)
Comment (ajaybhargavb) [inline src/NuGetPackageVerifier/Program.cs]: Nvm. Just saw your (Fix NRE) comment

File: src/NuGetPackageVerifier/Rules/DotNetCliToolPackageRule.cs [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+using System.Linq;
+using NuGet.Frameworks;
+using NuGet.Packaging;
+using NuGet.Packaging.Core;
+
+namespace NuGetPackageVerifier.Rules
+{
+    public class DotNetCliToolPackageRule : IPackageVerifierRule
+    {
+        private static readonly NuGetFramework _expectedFramework = FrameworkConstants.CommonFrameworks.NetCoreApp10;
```
Comment (ajaybhargavb) [inline src/NuGetPackageVerifier/Rules/DotNetCliToolPackageRule.cs]: I have no context here but don't we use netcoreapp1.1?

Comment (natemcmaster) [inline src/NuGetPackageVerifier/Rules/DotNetCliToolPackageRule.cs]: Not for now. We are are required to use netcoreapp1.0.

File: src/NuGetPackageVerifier/CompositeRules/DefaultCompositeRule.cs (hunk 2 above)
Comment (natemcmaster) [inline src/NuGetPackageVerifier/CompositeRules/DefaultCompositeRule.cs]: Will do.

File: src/NuGetPackageVerifier/Program.cs [hunk 4]
```diff
@@ -75,33 +74,23 @@ public static int Main(string[] args)
...
-                new RequiredPackageMetadataRule(),
-                new SatellitePackageRule(),
-                new StrictSemanticVersionValidationRule(),
-                //Composite rules
-                new AdxVerificationCompositeRule(),
-                new DefaultCompositeRule(),
-                new NonAdxVerificationCompositeRule(),
-                new SigningVerificationCompositeRule(),
-            }.ToDictionary(t => t.GetType().Name, t => t);
+            var allRules = typeof(Program)
```
Comment (Eilon) [inline src/NuGetPackageVerifier/Program.cs]: Cool!

File: src/NuGetPackageVerifier/Rules/AssemblyIsBuiltInReleaseConfigurationRule.cs [hunk 5]
```diff
@@ -6,11 +6,20 @@
 using System.Linq;
 using Mono.Cecil;
 using Mono.Collections.Generic;
+using NuGetPackageVerifier.Logging;
 
 namespace NuGetPackageVerifier.Rules
 {
     public class AssemblyIsBuiltInReleaseConfigurationRule : AssemblyHasAttributeRuleBase
     {
+        // TODO remove
```
Comment (Eilon) [inline src/NuGetPackageVerifier/Rules/AssemblyIsBuiltInReleaseConfigurationRule.cs]: So what's the deal with this here?

Comment (natemcmaster) [inline src/NuGetPackageVerifier/Rules/AssemblyIsBuiltInReleaseConfigurationRule.cs]: https://github.com/aspnet/BuildTools/issues/140

Comment (Eilon) [inline src/NuGetPackageVerifier/Rules/AssemblyIsBuiltInReleaseConfigurationRule.cs]: O I C, thanks!
